src/button/component.jsx

```diff
@@ -892,17 +892,60 @@ export const Button : Component<ButtonOptions> = create({
             type:     'function',
             required: false,
             get value() : Function {
+                let initialHeight;
+                const logInlineGuestOutOfViewPortOnlyOnce = once((data) => {
+                    info('buttons_expansion_outside_viewport', data);
+                    immediateFlush();
+                });
+
                 return function onResizeHandler() {
                     const container = this.container;
-                    const parentContainer = this.container.parentElement;
+                    if (!initialHeight) {
+                        initialHeight = container.offsetHeight;
+                    }
 
-                    if (container && parentContainer && container.offsetHeight > parentContainer.offsetHeight) {
-                        info(`button_taller_than_parent`, {
-                            height:       container.offsetHeight,
-                            parentHeight: parentContainer.offsetHeight
-                        });
-                        flushLogs();
+                    const getScrollOffsetY = () => {
+                        if (window.pageYOffset) {
+                            return window.pageYOffset;
+                        }
+
+                        if (document.documentElement) {
+                            return document.documentElement.scrollTop;
+                        }
+
+                        return 0;
+                    };
+
+                    // explanation https://github.com/paypal/paypal-checkout-components/pull/1136#discussion_r298025574
+                    const checkIfExpansionInViewport = () => {
```

Comment:
Could this, and `getScrollOffsetY` be abstracted, and put into `lib/dom.js`? That way we can keep this wrapper thin and we can re-use the logic for any other components that need it.

Reply:
I will put the `getScrollOffsetY` into `lib/dom.js` and will refactor the `checkIfExpansionInViewport` later since the latter function needs some variables of the current scope and little hard to decouple those.